### ISSUE TITLE
Weird loads on trucks

### ISSUE THREAD

**Issue by Quaffek** [MaFi-Games/Captain-of-Industry-issues#614]
### Issue description

Sometimes when an order is given to empty out a storage building (solid and loose (haven't tested this with fluids yet)) multiple trucks will on transport one unit of cargo to a destination.

### Expected behavior

Instead of 60 trucks picking up on item, i expected one truck to transport 60 units in one go.

### Reproduction steps

1. in the attached savegame i switched my iron and steel production to a new area.
2. Deconstruct a filled storage, whilst having an empty one ready
3. and see for yourself

### Game version

v0.4.8

### Attachments

[new4.zip](https://github.com/MaFi-Games/Captain-of-Industry-issues/files/9172180/new4.zip)

**Comment by zuff-coi** [comment]
Because of priorities, if a storage is set to empty or full, trucks will ignore rules like "no partial loads" and will always fulfill the need of the storage. If the storage is 1 over, a truck will come grab it. 

This is working as intended and a way around it is to make sure you have another storage with enough space for a bigger truck load to drop off/pick up from.

Let me know if that makes sense to you!

**Comment by FioraAeterna** [comment]
I'm not sure this is entirely true as described. I noticed that there seems to be two distinct sets of behavior, one for setting a storage to "empty", and the other for deconstructing. If I set a storage to "empty", the trucks will often efficiently remove 20 or 60 at a time. _But if I *deconstruct* a storage, the trucks will suddenly switch to taking 1 or 2 at a time._ I can avoid the problem by emptying before deconstructing, but this is strange and nonintuitive behavior, IMO.

My wild guess is that it's somehow a workaround in the code for forcing the trucks to empty out the last trickle of items in a nearly-empty storage in order to make deconstruction work.
